syntax_and_semantics/operators.md

```diff
@@ -1,18 +1,42 @@
 # Operators
 
-Operators like `+` and `-` are regular method calls. For example:
+Crystal supports a number of operators.
 
-```crystal
-a + b
-```
+Almost all operators are actually method calls. For example `a + b` is
+semantically equivalent to `a.+(b)`, a call to method `+`
+on `a` with argument `b`.
 
-is the same as:
+There are however two special rules regarding operators syntax:
 
-```crystal
-a.+(b)
-```
+* The dot (`.`) between receiver and method name (i.e. the *operator*) can be omitted.
+* Chained operators are modified to enforce operator precedence.
+
+Enforcing operator precedence makes sure that an expression such as `1 * 2 + 3 * 4`
+is parsed as `(1 * 2) + (2 * 3)` to honour regular math rules.
+
+Regular method names must start with a letter, but operators only consist of
+special characters. Any method not starting with a letter is an operator method,
+but available operators are whitelisted in the compiler. It contains a list of
+method names considered operators and their precedence rules.
+
+Operators are implemented like any regular method, and the standard library
+has many of them for math and other purposes.
+
+## Declaring operator methods
+
+Most operators can be implemented as operator methods.
+
+One can assign any meaning to the operators, but it is advisable to stay within
+similar semantics to the generic operator meaning to avoid cryptic code that is
+confusing and behaves unexpectedly.
 
-You can define an operator for a type like this:
+### Unary operators
+
+Unary operators have only a single operand. Thus, a method implementation has no
+arguments and only operates on the first operand, the receiver `self`.
+
+The following example demonstrates the `Vector2` type as a two-dimensional vector
```

Comment:
Why is there now an example only for defining a unary operator but not for defining a binary operator?